02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -170,26 +170,158 @@ Here's what has been added to the above since then:
 
 ## Why is PureScript's Documentation Lacking and How Do We Improve It?
 
-- Philosophy of the language:
-    - avoid (success at all costs)
-    - wrong intepretation:
-    - right interpretation
-- Current priority of core contributors: adding powerful features via breaking changes
-    - wrong interpretation: they don't care
-    - right interpretation: want to improve language
-    - Why a `v1.0` hasn't been released
-- Core Contributors are spread thin, so they cannot respond quickly to things
-    - The '75 libraries' problem
-    - Entrusting library maintenance to others and supporting them (e.g. the `event-stream` incident)
-    - Just like reviewing code, reviewing documentation takes a lot of thought
-    - Hard to keep documentation momentum
-    - Which libraries are 'core libraries'?
-- No clearly-defined mutually-held community-wide goal
-    - A desire to help exists, but no one is coordinating resources strategically
-- What exactly are FP's "Best Practices?"
-    - They are not well-defined, are assumed, or are habits
-    - Knowledge silos exist, so it's hard to extract that info
-- Slack-based discussions do not persist
+These are the reasons. Each wil be covered more in-depth following this:
+
+- PureScript is not _currently_ trying to be the next "mainstream language".
+- Core contributors are spread thin: limited time with too much responsibility
+- There is no roadmap that coordinates efforts
+- It's difficult to pin down and define what are Functional Programming's "best practices"
```

Comment:
Perhaps this idea is better phrased as a more general problem.
Perhaps:
- "Avoidance of prescribing best practices" (perhaps because the community is so humble? or just inexperienced?)
- "No unifying set of principles" (perhaps intentional, because the compiler team wants it to be unopinionated)
- "The lines between language principles, personal preferences, and best practices is undefined" (This feels right, but I haven't thought deeply enough to settle on this being true. Also, I'd need to find examples for each of these three categories.)

Reply:
> "The lines between language principles, personal preferences, and best practices is undefined" (This feels right, but I haven't thought deeply enough to settle on this being true. Also, I'd need to find examples for each of these three categories.)

Honestly, this seems like a great way to word this. Perhaps replace "undefined" with "unclear" / "foggy". Or perhaps say they are overlapping?
For example, is a Monad a design pattern? I don't initially see it as such, but I guess it could be called that. At the same time, how else would you do something in FP without Monads?
Are phantom types design patterns? Should one always use smart constructors?

It's like there are things that we just take for granted.

Reply:
I want to push some changes that address this, but I'm not sure what to say yet. Shall we continue this discussion on the PR thread rather than in this specific comment thread?